### DISCUSSION TITLE
Test set for dataflow

### DISCUSSION THREAD

**Discussion by popthink** [plume-oss/plume#146] [Ideas]
Now Plume can produce dataflow edges.

For such things, Test set may be enhanced.

There is some way to make it.

1. Using algorithm of soot(Reaching Def)

2. Just checking them with "should ~"

**Comment by popthink** [comment]
Option 1 is not worth. Because of the comment @DavidBakerEffendi wrote in #127.
Soot doesn't over-approximate and let them tainted on every call.
Option 2 might be only option for checking data flow working properly.
Unless plume provide some own dataflow pass based on soot or something for cpg.

**Reply by DavidBakerEffendi** [reply]
So this might be a decent place to add another test set under the `testing` submodule since then we can use the `Cpg` API that is only available for Scala - which allows better graph traversals. I guess validating some small examples with basic dependencies should be good enough?

**Reply by popthink** [reply]
I agree :+1:

**Reply by DavidBakerEffendi** [reply]
Created an issue under #160
